centralizedmetrics/sqlite_persistence.go

```diff
@@ -0,0 +1,201 @@
+package centralizedmetrics
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/status-im/status-go/centralizedmetrics/common"
+)
+
+type SQLiteMetricRepository struct {
+	db *sql.DB
+}
+
+func NewSQLiteMetricRepository(db *sql.DB) *SQLiteMetricRepository {
+	return &SQLiteMetricRepository{db: db}
+}
+
+func (r *SQLiteMetricRepository) Poll() ([]common.Metric, error) {
+	tx, err := r.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	userID, err := r.UserID(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := tx.Query("SELECT id, event_name, event_value, platform, app_version, timestamp FROM centralizedmetrics_metrics limit 10")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var metrics []common.Metric
+	for rows.Next() {
+		var metric common.Metric
+		var eventValue string
+
+		if err := rows.Scan(&metric.ID, &metric.EventName, &eventValue, &metric.Platform, &metric.AppVersion, &metric.Timestamp); err != nil {
+			return nil, err
+		}
+
+		// Deserialize eventValue
+		if err := json.Unmarshal([]byte(eventValue), &metric.EventValue); err != nil {
+			return nil, err
+		}
+
+		metric.UserID = userID
+
+		metrics = append(metrics, metric)
+	}
+
+	return metrics, rows.Err()
+}
+
+func (r *SQLiteMetricRepository) Delete(metrics []common.Metric) error {
+	tx, err := r.db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	stmt, err := tx.Prepare("DELETE FROM centralizedmetrics_metrics WHERE id = ?")
```

Comment:
Wouldn't it be better to use `IN` to delete all metrics at once? Especially because we're processing them in small batches it should be fine.

Reply:
it's probably faster, yes, but if it's ok I would address separately